src/core/matcher.js

```diff
@@ -0,0 +1,18 @@
+exports.registerChaiAssertion = (chai, utils) => {
```

Comment:
These should be tested in unit tests, plus I'm not sure `assert` take that much parameters. Aren't only the first two used ?

Reply:
WDYT about renaiming this file to `custom_chai_assertions` ?

Reply:
Yeah, I definitely should have tested that bit and I'm fine with the rename too.

As for the `assert` method, I think this is a "private" one that differs from the one exposed as the assertion API (cf. [Adding method](https://www.chaijs.com/guide/helpers/#adding-methods)).